google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -167,23 +168,29 @@ public Project getProject() {
       return Optional.of(createStandardRunner(loggingHandler, Paths.get(source.getFilePath()),
           deploy, isFlexCompat));
     } else if (targetEnvironment.isFlexible()) {
-      // Checks if the Yaml or Dockerfile exist.
-      Optional<FlexibleRuntime> runtimeOptional =
-          AppEngineProjectService.getInstance().getFlexibleRuntimeFromAppYaml(
-              deploymentConfiguration.getAppYamlPath());
-
-      if (!Files.exists(Paths.get(deploymentConfiguration.getAppYamlPath()))) {
-        callback.errorOccurred(GctBundle.getString("appengine.deployment.error.staging.yaml"));
-        return Optional.empty();
-      }
-      if (runtimeOptional.filter(runtime -> runtime == FlexibleRuntime.custom).isPresent()
-          && !Files.exists(Paths.get(deploymentConfiguration.getDockerFilePath()))) {
+      try {
+        // Checks if the Yaml or Dockerfile exist.
+        Optional<FlexibleRuntime> runtimeOptional =
+            AppEngineProjectService.getInstance().getFlexibleRuntimeFromAppYaml(
+                deploymentConfiguration.getAppYamlPath());
+
+        if (!Files.exists(Paths.get(deploymentConfiguration.getAppYamlPath()))) {
+          callback.errorOccurred(GctBundle.getString("appengine.deployment.error.staging.yaml"));
+          return Optional.empty();
+        }
+        if (runtimeOptional.filter(runtime -> runtime == FlexibleRuntime.custom).isPresent()
+            && !Files.exists(Paths.get(deploymentConfiguration.getDockerFilePath()))) {
+          callback.errorOccurred(
+              GctBundle.getString("appengine.deployment.error.staging.dockerfile"));
+          return Optional.empty();
+        }
+        return Optional.of(createFlexRunner(loggingHandler, Paths.get(source.getFilePath()),
+            deploymentConfiguration, deploy));
+      } catch (MalformedYamlFileException myf) {
         callback.errorOccurred(
-            GctBundle.getString("appengine.deployment.error.staging.dockerfile"));
+            GctBundle.message("appengine.appyaml.malformed") + "\n" + myf.getMessage());
```

Comment:
This gets shown, for example, when a valid run configuration is saved, the app.yaml file is made invalid and, immediately after, there is an attempt to run that run configuration. It gets shown on the log window.

I thought it would be nice to be more helpful, even though this case should be pretty unusual. It would be better to show it, e.g., in the deployment or facet UI, but I thought the text got too big.

The exception message basically gives more detail around where the yaml file is invalid, which can be helpful in large files. But the same can be accomplished with a YAML file editor.